src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1930,6 +1930,29 @@ public ErrorDiagnostic RuntimeValueNotAllowedInFunctionDeclaration(string? acces
                 TextSpan,
                 "BCP343",
                 $@"Using a func declaration statement requires enabling EXPERIMENTAL feature ""{nameof(ExperimentalFeaturesEnabled.UserDefinedFunctions)}"".");
+        
+            public ErrorDiagnostic ExpectBodyStart() => new(
+                TextSpan,
+                "BCP345",
+                "Expected the \"{\" character at this location.");
+            public ErrorDiagnostic TestDeclarationMustReferenceBicepTest() => new(
+                TextSpan,
+                "BCP346",
+                "A Test declaration can only reference a Bicep File, an ARM template, a registry reference or a template spec reference.");
```

Comment:
Will test files be standard templates, or are they going to have a different grammar/file extension? The "an ARM template, a registry reference or a template spec reference" clause implies that tests will be persisted in the compiled JSON of a template -- is that going to be the case?